gst/nnstreamer/nnstreamer_plugin_api_filter.h

```diff
@@ -111,38 +111,106 @@ typedef struct _GstTensorFilterProperties
+{
+  uint64_t version; /**< Version of the struct, can be 0x0 (earlier version (_GstTensorFilterFramework_v1)) or 0x2 (newer version (_GstTensorFilterFramework_v2)) */
+
+  int (*open) (const GstTensorFilterProperties * prop, void **private_data);
+  /**< Optional. tensor_filter.c will call this before any of other callbacks and will call once before calling close
+   *
+   * @param[in] prop read-only property values
+   * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, open() allocates memory for private_data.
+   * @return 0 if ok. < 0 if error.
+   */
+
+  void (*close) (const GstTensorFilterProperties * prop, void **private_data);
+  /**< Optional. tensor_filter.c will not call other callbacks after calling close. Free-ing private_data is this function's responsibility. Set NULL after that.
+   *
+   * @param[in] prop read-only property values
+   * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer. Normally, close() frees private_data and set NULL.
+   */
 
   int (*invoke_NN) (const GstTensorFilterProperties * prop, void **private_data,
       const GstTensorMemory * input, GstTensorMemory * output);
-      /**< Mandatory callback. Invoke the given network model.
-       *
-       * @param[in] prop read-only property values
-       * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer.
-       * @param[in] input The array of input tensors. Allocated and filled by tensor_filter/main
-       * @param[out] output The array of output tensors. Allocated by tensor_filter/main and to be filled by invoke_NN. If allocate_in_invoke is TRUE, sub-plugin should allocate the memory block for output tensor. (data in GstTensorMemory)
-       * @return 0 if OK. non-zero if error.
-       */
+  /**< Mandatory callback. Invoke the given network model.
+   *
+   * @param[in] prop read-only property values
+   * @param[in/out] private_data A subplugin may save its internal private data here. The subplugin is responsible for alloc/free of this pointer.
+   * @param[in] input The array of input tensors. Allocated and filled by tensor_filter/main
+   * @param[out] output The array of output tensors. Allocated by tensor_filter/main and to be filled by invoke_NN. If allocate_in_invoke is TRUE, sub-plugin should allocate the memory block for output tensor. (data in GstTensorMemory)
+   * @return 0 if OK. non-zero if error.
+   */
 
-  int (*getInputDimension) (const GstTensorFilterProperties * prop,
-      void **private_data, GstTensorsInfo * info);
+  /**
+   * @brief Distinct elements between two versions of the subplugin interfaces
+   */
+  union
+  {
+    /**
+     * @brief Tensor_Filter Subplugin definition Version 0
+     */
+    struct /** _GstTensorFilterFramework_v1 */
```

Comment:
s/v1/v0/

Reply:
done `s/Version 1/Version 0/` and `s/Version 2/Version 1/`